### ISSUE TITLE
Different permissions for draft editing and approval

### ISSUE THREAD

**Issue by andreabz** [cosmocode/dokuwiki-plugin-structpublish#15]
I would suggest keeping separate permissions for editing drafts and page approval.
Currently, as described in #5, to edit drafts you need both ACL edit permissions and structpublish approval permissions, otherwise you won't see the newly created draft.
It would be nice to have a new separate structpublish see-and-edit draft permission in the Structured publish data section, so that:

- to edit a draft of a published page, a user would need both ACL edit permissions and structpublish see-and-edit drafts permissions;
- to approve a draft, a user would need both ACL edit permissions and structpublish approval permissions;
- to publish and approved draft, a user would need both ACL edit permissions and structpublish publish permissions.

This feature would also help in workflows where editors are not in charge of approving or publishing pages.

**Comment by annda** [comment]
This is a duplicate of #5 and should be a comment in #5. Out of convenience and only because this issue is more elaborate, I will make an exception and close the original issue.

**Comment by annda** [comment]
I agree that we need to separate more clearly the approval permissions and the "see-and-edit drafts" permissions. The latter is IMO identical to ACLs, so lets not duplicate that but just check for ACLs in the appropriate places (I'm thinking of `show` and `banner` actions). It also makes sense to give editors (as determined by ACLs) a button to mark their draft as "ready for approval". This button could then trigger a notification to approvers.

I'm not yet sure what the best way to implement it would be. Probably adding another status, something like`final_draft`.

@splitbrain @andreabz  @moka-dev Any thoughts?

**Comment by splitbrain** [comment]
Quick summary of a in-person discussion between me and @annda 

1. Users with write permissions are allowed to see and edit drafts when accessing the page. _This is new and needs to be implemented_
2. Publishers can see drafts when accessing the page. They do not necessarily need to have write permissions. _We need to check if the latter is already implemented correctly here_ 
3. In struct aggregations, we currently only show draft data to users that are approvers/publishers.  _I would argue that this is sensible and does not need changing. Others might feel differently._
4. Marking and notifying about a draft being ready for approval is a different issue and should be discussed separately. _Having an arbitrary amount of steps and allowing for binding approval permissions to ACLs might be a good solution here_

**Comment by annda** [comment]
1. Editors should see drafts even if they have no role - implemented in [9cc4b3d](https://github.com/cosmocode/dokuwiki-plugin-structpublish/commit/9cc4b3d87542b515048cd687b00862894c1cb6c7)
2. Publishers without write permissions should see drafts - this is working as expected.

**Comment by moka-dev** [comment]
> I agree that we need to separate more clearly the approval permissions and the "see-and-edit drafts" permissions. The latter is IMO identical to ACLs, so lets not duplicate that but just check for ACLs in the appropriate places (I'm thinking of `show` and `banner` actions). It also makes sense to give editors (as determined by ACLs) a button to mark their draft as "ready for approval". This button could then trigger a notification to approvers.
> 
> I'm not yet sure what the best way to implement it would be. Probably adding another status, something like`final_draft`.
> 
> [@splitbrain](https://github.com/splitbrain) [@andreabz](https://github.com/andreabz) [@moka-dev](https://github.com/moka-dev) Any thoughts?

This would be great!
